Source/WebInspectorUI/UserInterface/Controllers/ConsoleManager.js

```diff
@@ -88,6 +109,33 @@ WI.ConsoleManager = class ConsoleManager extends WI.Object
         this._remoteObjectsToRelease.add(remoteObject);
     }
 
+    addSnippet(snippet)
+    {
+        console.assert(snippet instanceof WI.ConsoleSnippet, snippet);
+        console.assert(!this._snippets.has(snippet), snippet);
+        console.assert(!this._snippets.some((existingSnippet) => snippet.contentIdentifier === existingSnippet.contentIdentifier), snippet);
+
+        this._snippets.add(snippet);
+
+        if (!this._restoringSnippets)
+            WI.objectStores.consoleSnippets.putObject(snippet);
+
+        this.dispatchEventToListeners(WI.ConsoleManager.Event.SnippetAdded, {snippet});
+    }
+
+    removeSnippet(snippet)
+    {
+        console.assert(snippet instanceof WI.ConsoleSnippet, snippet);
+        console.assert(this._snippets.has(snippet), snippet);
+
+        this._snippets.delete(snippet);
+
+        if (!this._restoringSnippets)
```

Comment:
Possible cleanup. This could probably be an assert. I don't think we'd ever expect a remove during restoration? `console.assert(!this._restoringSnippets)`. But it is nice to be symmetric with `addSnippet`.

Reply:
I suppose someone could technically remove a snippet in response to `WI.ConsoleManager.Event.SnippetAdded` during restoration... But I like the assertion because we'd never expect that.